### PR TITLE
Fix for long length SPI writes on ATMEL

### PR DESCRIPTION
**Pull request by gamblor21**
Fix for #7670

Writes over 64Kb were failing as that is the maximum a DMA write can handle.

Possible other fix would involve chaining DMA transaction buffers but the break between buffers was about 3 microseconds so probably not an issue.

This could be moved into `dma.c` but then would have to be careful about QSPI (which has this problem but at 4x the size). This was the easier place to put it in, but open if we want to modify it too. It seems only SPI/QSPI uses the routine in question.

### CODE REVIEW

**Review by jepler — APPROVED** [review]
Looks pretty safe, though I notice that for (say) a transfer of 65536+16 bytes it will do a DMA transfer of the final 16 bytes. This is probably just fine, I think avoiding DMA for small transfers is just for efficiency, not correctness.  Didn't test, would love to hear from @gamblor21 before merging if this fixes the original problem

**Comment by jepler** [comment]
oh wait, you filed the PR.  I'm apparently not far enough into this mug of coffee yet :)

**Review by dhalbert — COMMENTED** [review]
One comment.

File: ports/atmel-samd/common-hal/busio/SPI.c [hunk 1]
```diff
@@ -268,7 +268,14 @@ bool common_hal_busio_spi_write(busio_spi_obj_t *self,
...
     if (len >= 16) {
-        status = sercom_dma_write(self->spi_desc.dev.prvt, data, len);
+        size_t bytes_remaining = len;
+
+        // Maximum DMA transfer is 65535
+        while (bytes_remaining > 0) {
+            size_t to_send = (bytes_remaining > 65535) ? 65535 : bytes_remaining;
+            status = sercom_dma_write(self->spi_desc.dev.prvt, data + (len - bytes_remaining), to_send);
+            bytes_remaining -= to_send;
+        }
```
[the remark this answers is not in the document]
Comment (dhalbert) [reply ports/atmel-samd/common-hal/busio/SPI.c:278]: addressed in #7700

**Pull request merged**